src/Microsoft.IdentityModel.Xml/IssuerSerial.cs

```diff
@@ -53,27 +54,24 @@ public IssuerSerial(string issuerName, string serialNumber)
             SerialNumber = serialNumber;
         }
 
-        /// <summary>
-        /// Compares two IssuerSerial objects.
-        /// </summary>
+        /// <inheritdoc/>
         public override bool Equals(object obj)
         {
-            var other = obj as IssuerSerial;
-            if (other == null)
-                return false;
-            else if (string.Compare(IssuerName, other.IssuerName, StringComparison.OrdinalIgnoreCase) != 0
-                || string.Compare(SerialNumber, other.SerialNumber, StringComparison.OrdinalIgnoreCase) != 0)
-                return false;
-            return true;
+            return obj is IssuerSerial serial &&
+                string.Equals(IssuerName, serial.IssuerName, StringComparison.OrdinalIgnoreCase) &&
+                string.Equals(SerialNumber, serial.SerialNumber, StringComparison.OrdinalIgnoreCase);
         }
 
-        /// <summary>
-        /// Serves as a hash function for IssuerSerial.
-        /// </summary>
+        /// <inheritdoc/>
         public override int GetHashCode()
         {
-            return base.GetHashCode();
+            unchecked
+            {
+                int hashCode = -1073543679;
```

Comment:
I'm guessing that this code is emitted by VS, right?

Reply:
Correct. It didn't originally add `unchecked` though.

Reply:
@keegan-caruso what is the purpose of the multiplication step, how is that different than a fix number?
With the fixed number, what if two objects have the same "IssuerName" and use the same 'magic' number?

Reply:
This was generated code. Honestly, was just relying on the tool to create what we hope is a good distribution. 

And yes, I believe a follow up call (Equals) will always be required. HashCode should not be used to determine equality.